File: C/Par/saxpy_openmp.c
#include <omp.h>

void saxpy(int n, float a, float *x, float *y)
{
    #pragma omp parallel for
    for (int i = 0; i < n; i++)
    {
        y[i] = a * x[i] + y[i];
    }
}

File: Fortran/3.5/saxpy_openmp.f90
subroutine saxpy_parallel(n, a, x, y)
    implicit none
    integer, intent(in) :: n
    real, intent(in) :: a
    real, intent(in) :: x(n)
    real, intent(inout) :: y(n)
    integer :: i

    !$OMP PARALLEL DO
    do i = 1, n
        y(i) = a * x(i) + y(i)
    end do
    !$OMP END PARALLEL DO

end subroutine saxpy_parallel
